tests/cli.rs

```diff
@@ -0,0 +1,198 @@
+use std::process::Command;
+
+use assert_cmd::prelude::*;
+use predicates::prelude::*;
+
+#[test]
+fn empty_command_args() -> Result<(), Box<dyn std::error::Error>> {
+  let mut cmd = Command::cargo_bin("tailcall")?;
+
+  cmd.arg("");
+  cmd
+    .assert()
+    .failure()
+    .stderr(predicate::str::contains("Usage: tailcall"));
+
+  Ok(())
+}
+
+// Check command tests
+#[test]
+fn check_command_file_not_specified() -> Result<(), Box<dyn std::error::Error>> {
+  let mut cmd = Command::cargo_bin("tailcall")?;
+
+  cmd.arg("check");
+  cmd.assert().failure().stderr(predicate::str::contains(
+    "error: the following required arguments were not provided",
+  ));
+  cmd
+    .assert()
+    .failure()
+    .stderr(predicate::str::contains("Usage: tailcall check <FILE_PATH>"));
+
+  Ok(())
+}
+
+#[test]
+fn check_command_file_doesnt_exist() -> Result<(), Box<dyn std::error::Error>> {
+  let mut cmd = Command::cargo_bin("tailcall")?;
+
+  cmd.arg("check").arg("test.file.doesnt.exist.graphql");
+  cmd
+    .assert()
+    .failure()
+    .stderr(predicate::str::contains("Error: No such file or directory"));
+
+  Ok(())
+}
+
+#[test]
+fn check_command_file_exists_and_invalid() -> Result<(), Box<dyn std::error::Error>> {
+  let mut cmd = Command::cargo_bin("tailcall")?;
+
+  cmd
+    .arg("check")
+    .arg("tests/graphql/errors/test-const-with-inline.graphql");
+  cmd
+    .assert()
+    .failure()
+    .stderr(predicate::str::contains("Error: Validation Error"));
+
+  Ok(())
+}
+
+#[test]
+fn check_command_file_exists_and_valid() -> Result<(), Box<dyn std::error::Error>> {
+  let mut cmd = Command::cargo_bin("tailcall")?;
+
+  cmd
+    .arg("check")
+    .arg("examples/jsonplaceholder.graphql")
+    .arg("--n-plus-one-queries")
+    .arg("--schema");
+  cmd
+    .assert()
+    .success()
+    .stdout(predicate::str::contains("No errors found"));
+
+  Ok(())
+}
+
+#[test]
+fn check_command_file_exists_and_valid2() -> Result<(), Box<dyn std::error::Error>> {
+  let mut cmd = Command::cargo_bin("tailcall")?;
+  cmd
+    .arg("check")
+    .arg("examples/jsonplaceholder.graphql")
+    .arg("-n")
+    .arg("-s");
+  cmd
+    .assert()
+    .success()
+    .stdout(predicate::str::contains("No errors found"));
+
+  Ok(())
+}
+
+#[test]
+fn check_command_file_exists_and_valid3() -> Result<(), Box<dyn std::error::Error>> {
+  let mut cmd = Command::cargo_bin("tailcall")?;
+  cmd.arg("check").arg("examples/jsonplaceholder.graphql").arg("-s");
+  cmd
+    .assert()
+    .success()
+    .stdout(predicate::str::contains("No errors found"));
+
+  Ok(())
+}
+
+// Start command tests
```

Comment:
we can wrap several related tests in `mod` and it will be easier to see the test run results and filter test run based on the command name